core/src/main/java/org/apache/carbondata/core/metadata/SegmentFileStore.java

```diff
@@ -1125,19 +1125,21 @@ public static void deleteSegment(String tablePath, Segment segment,
     SegmentFileStore fileStore = new SegmentFileStore(tablePath, segment.getSegmentFileName());
     List<String> indexOrMergeFiles = fileStore.readIndexFiles(SegmentStatus.SUCCESS, true,
         FileFactory.getConfiguration());
-    Map<String, List<String>> indexFilesMap = fileStore.getIndexFilesMap();
```

Comment:
If the segment is an external segment, then in that case in the segment file of the segment isRelative is false, i think it's better to check and skip it this way. What do you suggest @QiangCai ? @ajantha-bhat